components/profile/activityTab/HistoryRow.vue

```diff
@@ -135,25 +135,23 @@
 
 <script setup lang="ts">
 import { NeoAvatar, NeoTooltip } from '@kodadot1/brick'
-import { parseNftAvatar } from '@/utils/nft'
 import { Event } from './History.vue'
 import {
   blank,
   interactionNameMap,
 } from '@/components/collection/activity/events/eventRow/common'
 import EventTag from '@/components/collection/activity/events/eventRow/EventTag.vue'
 import BlockExplorerLink from '@/components/shared/BlockExplorerLink.vue'
-
+import { useGalleryItem } from '@/components/gallery/useGalleryItem'
 const props = defineProps<{
   event: Event
   withToColumn: boolean
   variant: 'Desktop' | 'Touch'
 }>()
 
+const { nftResources, nftImage } = useGalleryItem(props.event.Item.id)
```

Comment:
nope.
this will make a graphql call for each row, where you should already get all the data you need in the call that fetches the events

![image](https://github.com/kodadot/nft-gallery/assets/22791238/b56f3213-7d51-4e28-8148-79d177cbde0b)

Reply:
Some of your nfts can't get their image urls with `processSingleMetadata`  method in `nft.parseNftAvatar`. Each of them has  more than one image, for example [Referendum 209](https://kodadot.xyz/ksm/gallery/13485165-3208723ec6f65df810-ITEM-209ST-00000317) . This api responds with 
```
"'Stamp'

In Referendum 209 the community voted on whether to reset the Pichiu Parachain in order to fix bugs. The underlying proposal of this Referendum has been replaced by Referendum 205. Since Referendum 205 has passed community vote already, this Referendum was superfluous.

Artist: @kseniashikh

Type of vote: Direct."
```
I have to get their images by graphql call. It's better to  fix this abnormal response so that we can keep efficient.

Reply:
see `useGalleryItem` line 113
`nftMetadata.value = await getNftMetadata(nftEntity, urlPrefix.value)`

uses `getNftMetadata`

this is in contrast to `parseNftAvatar` that uses `processSingleMetadata`

that's the direction I would look at

Reply:
`parseNftAvatar` call stack:
 - processSingleMetadata

`getNftMetadata` call stack:
- getProcessMetadata
  - processSingleMetadata

they both come to the same method at last.

Reply:
let's look at `getNFTMetadata`:
![image](https://github.com/kodadot/nft-gallery/assets/22791238/3d66e534-133a-407b-922b-d5938261063b)

where do we have problem with profile activity images? on rmrk2
![image](https://github.com/kodadot/nft-gallery/assets/22791238/d34b734a-9768-48e5-b27c-c66d73ab4456)


so in this case we need the resources field on the nft.

but wait , the query to get the events doesn't fetch resources field:

```graphql
query allEventsByProfile($id: String) {
  events(where: {caller_eq: $id, OR: {currentOwner_eq: $id}}) {
    meta
    interaction
    blockNumber
    timestamp
    currentOwner
    caller
    id
    nft {
      id
      name
      metadata
      collection {
        id
        name
        __typename
      }
      __typename
    }
    __typename
  }
}
```

so what need to be done?

1. make the query fetch resources
2.  use `getNftMetadata`  from `useNFT` composable

Reply:
Please don't resolve conversations that reviewers started
It is up to me to do so

Reply:
mb. I don't do this before